services/secret-service/server/src/main/java/org/apache/airavata/mft/secret/server/backend/file/FileBasedSecretBackend.java

```diff
@@ -170,4 +170,46 @@ public boolean updateAzureSecret(AzureSecretUpdateRequest request) throws Except
     public boolean deleteAzureSecret(AzureSecretDeleteRequest request) throws Exception {
         throw new UnsupportedOperationException("Operation is not supported in backend");
     }
+
+    @Override
+    public Optional<GCSSecret> getGCSSecret(GCSSecretGetRequest request) throws Exception {
+        JSONParser jsonParser = new JSONParser();
+        InputStream inputStream = FileBasedSecretBackend.class.getClassLoader().getResourceAsStream(secretFile);
+
+        try (InputStreamReader reader = new InputStreamReader(inputStream)) {
+            Object obj = jsonParser.parse(reader);
+
+            JSONArray resourceList = (JSONArray) obj;
+
+            List<GCSSecret> gcsSecrets = (List<GCSSecret>) resourceList.stream()
+                    .filter(resource -> "GCS".equals(((JSONObject) resource).get("type").toString()))
+                    .map(resource -> {
+                        JSONObject r = (JSONObject) resource;
+
+                        GCSSecret gcsSecret = GCSSecret.newBuilder()
+                                .setSecretId(r.get("secretId").toString())
+                                .setJsonCredentialsFilePath(r.get("jsonCredentialsFilePath").toString()).build();
```

Comment:
We can't expect the JSON file path available in agent running location. This makes very hard to make the agent stateless. You should serialize json into the secret itself